packages/expo-updates/ios/EXUpdates/Logging/UpdatesLogReader.swift

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022-present 650 Industries. All rights reserved.
+
+// Reads expo-updates logs
+
+import Foundation
+import OSLog
+
+import ExpoModulesCore
+import SafariServices
+
+@available(macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0, *)
+@objc(EXUpdatesLogReader)
+public class UpdatesLogReader: NSObject {
+
+  @objc(getLogEntriesNewerThan:)
+  public func getLogEntries(newerThan: Date) -> NSArray {
+    return getLogEntries(newerThan: newerThan, returnEntriesAs: "NSDictionary")
+  }
+
+  /**
+   Get expo-updates logs newer than the given date
+   Returns an Objective-C NSArray of either strings or NSDictionary objects
+   Returned strings are all in the JSON format of UpdatesLogEntry
+   Maximum of one day lookback is allowed
+   */
+  @objc(getLogEntriesNewerThan:returnEntriesAs:)
+  public func getLogEntries(newerThan: Date, returnEntriesAs: String = "NSDictionary") -> NSArray {
+    let result = NSMutableArray()
+    do {
+      let earliestDate = Date().addingTimeInterval(-86400)
+      let dateToUse = newerThan.timeIntervalSince1970 < earliestDate.timeIntervalSince1970 ?
+      earliestDate :
+      newerThan
+
+      let logStore = try OSLogStore(scope: .currentProcessIdentifier)
+      // Get all the logs since the given date.
+      let position = logStore.position(date: dateToUse)
+
+      // Fetch log objects, selecting our subsystem and category
+      let predicate = NSPredicate(format: "category == %@ AND subsystem = %@",
+                                      argumentArray: ["expo-updates", "dev.expo.modules"])
+      let allEntries = try logStore.getEntries(at: position, matching: predicate)
+
+      // Extract just the log message strings
+      let allEntriesMessages = allEntries
+          .compactMap { $0 as? OSLogEntryLog }
+          .compactMap { $0.composedMessage }
+      for entry in allEntriesMessages.enumerated() {
+        let json = entry.element as String
+        if returnEntriesAs == "NSString" {
+          result.add(json)
+          continue
+        }
+        let logEntry = UpdatesLogEntry.create(from: json)
+        result.add(logEntry!.asDict())
+      }
+    } catch {
+      result.add("Error occurred in UpdatesLogReader: \(error.localizedDescription)")
```

Comment:
not sure this pattern of returning an error string as an arbitrary element will be consumable. Might make more sense to just throw the error?

This also is a bit messy for the dictionary element return type.

Reply:
Hmm I will think more on how to handle this better. No error should ever be thrown really, but I'll have a look at this.

Reply:
Completed this change